### ISSUE TITLE
Sole Treadmill F63

### ISSUE THREAD

**Issue by Smileyvero** [cagnulein/qdomyos-zwift#2712]
**Describe the bug**
The treadmill (Sole f63, 2016 model) doesn't connect to the app. The qz app says that the treadmill is found, but the stats in the app is zero. 

**To Reproduce**

**Expected behavior**
I want the treadmill to connect to the qz app, so I can use zwift. 

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Samsung S21
 - Version 14

**Append a debug log**
[inbound2062485633441022172.zip](https://github.com/user-attachments/files/17568362/inbound2062485633441022172.zip)

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**
My phone founds the treadmill with bluetooth (but its not connected together) . I have tried to use several android phones and tablet to see if there is something wrong with my phone, without any luck. I started the mill first, before i start the app. But nothing seems to work. I have tried to restart the phone and the treadmill, and it still dont work.

**Comment by cagnulein** [comment]
hi @Smileyvero did you try to enable the sole f63 setting under the settings, treadmill options, sole treadmill?
Let me know!

**Comment by Smileyvero** [comment]
> hi @Smileyvero did you try to enable the sole f63 setting under the settings, treadmill options, sole treadmill? Let me know!

Yes, i have tried that too, still wont work...

**Comment by cagnulein** [comment]
can you send me a debug log in this scenario?

**Comment by cagnulein** [comment]
also does the sole app works?

**Comment by Smileyvero** [comment]
[debug-Wed_Oct_30_09_46_09_2024.zip](https://github.com/user-attachments/files/17568709/debug-Wed_Oct_30_09_46_09_2024.zip)

The Sole+ app dosent work, but the Sole fitness app work, sometimes... 😅 I had luck with the mill and the qz app the first day i tried (couple of days ago) , but since then, nothing works.

**Comment by cagnulein** [comment]
@Smileyvero here you also set the f63 as a ftms treadmill. don't. please remove the f63 in the ftms treadmill settings and press ok 
and also enable the "Inclination (experimental)" under the sole settings and restart the app. let me know if it works, if doesn't create me another debug log
thanks

**Comment by Smileyvero** [comment]
[debug-Wed_Oct_30_10_04_08_2024.zip](https://github.com/user-attachments/files/17568896/debug-Wed_Oct_30_10_04_08_2024.zip)

Still dont work 😔

**Comment by cagnulein** [comment]
the treadmill is not talking at all. did you try to reboot it?

**Comment by Smileyvero** [comment]
Yes, have turned it off/on several times.... Have turned it off with the main switch, but also by unplugging the socket for a few seconds. That's what I was a little worried about, that the bluetooth on the treadmill doesn't work...

**Comment by cagnulein** [comment]
did you try now with the sole fitness app? because if you tried at the very begginning maybe something has changed? if it still works i can check the communication following this with this app https://github.com/cagnulein/qdomyos-zwift/blob/master/docs/20_supported_devices_and_applications.md#android-device

**Comment by Smileyvero** [comment]
Yes, the mill does connect with the Sole fitness app now, and work just fine with the app 😊 I am going to see if i can understand the progress in the link 😄👍

**Comment by cagnulein** [comment]
ok that's a good news because it means that the treadmill is fully functional. we need only to understand why it's not working with QZ

**Comment by cagnulein** [comment]
also in the meantime could you please also try this

https://github.com/cagnulein/qdomyos-zwift/actions/runs/11590424362

Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

**Comment by Smileyvero** [comment]
It doesn't seem quite seamless yet, it's a bit unstable. I therefore have a new bug report which I am attaching here now.

[debug-Wed_Oct_30_11_25_24_2024.zip](https://github.com/user-attachments/files/17570556/debug-Wed_Oct_30_11_25_24_2024.zip)

**Comment by cagnulein** [comment]
sorry i don't understand: you said before by email that it was working,, but now it doesn't anymore?

**Comment by cagnulein** [comment]
@Smileyvero do you have a log where the treadmill was working correctly with the qz app? so i can compare them

**Comment by Smileyvero** [comment]
No, unfortunately I haven't, I wasn't quite prepared for it to work so well so quickly 😬 But I'll test a bit and see if I can recreate when it worked so well.

**Comment by cagnulein** [comment]
ok thanks, let me know. maybe it was something about open qz when the treadmill was already running or something similar. let me know

**Comment by Smileyvero** [comment]
Here, at the end at the debug-log, it start working again. The treadmill was already running when it happens. 😊

[debug-Wed_Oct_30_11_53_14_2024.zip](https://github.com/user-attachments/files/17570981/debug-Wed_Oct_30_11_53_14_2024.zip)

**Comment by cagnulein** [comment]
perfect! let me check it!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 30 ott 2024 alle ore 11:56 Smileyvero <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
ok i found a difference, let me try to implement this

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 30 ott 2024 alle ore 12:00 cagnulein ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
@Smileyvero what i found so far: is it possible that you need to start the treadmill by the treadmill directly to start sending the metrics to qz? Did you check this?

**Comment by Smileyvero** [comment]
Now I have created a new debug log. I started the mill first, then connected to the app. It didn't work. Then I closed the app, opened it, and tried again. Then the mill starts to live its own life 🤷 It adjusts the speed completely by itself, without me pressing anything.

[debug-Wed_Oct_30_12_16_47_2024.zip](https://github.com/user-attachments/files/17571338/debug-Wed_Oct_30_12_16_47_2024.zip)

**Comment by Smileyvero** [comment]
But now i unfortunately have to run, I have a meeting to attend very soon. Thank you very much for your help so far, I will reply again as soon as I have a little time 😄

**Comment by cagnulein** [comment]
hi @Smileyvero did you do any other experiment?

**Comment by Smileyvero** [comment]
Thanks for bringing up the thread. It's been a bit busy lately, but I got to test it a bit against Zwift. It is unstable, and suddenly loses the bluetooth connection in the middle of zwifting. I don't think automatic incline works either. I have a debug log from that session, but it is possible that it is a bit long 😅

**Comment by cagnulein** [comment]
for automatic inclination did you set the zwift credentials in QZ?
for the bluetooth connection error let's check the debug log, just compress it before sending it.
Thanks

**Comment by Smileyvero** [comment]
[debug-Mon_Oct_28_11_52_14_2024.zip](https://github.com/user-attachments/files/17642322/debug-Mon_Oct_28_11_52_14_2024.zip)

Im not sure, but I think it's this one.... 😅

**Comment by cagnulein** [comment]
i guess not because i don't see any speed on this. Did you run on 28th of october? (the log is from that date) Otherwise send me all of them compressed in one file and i can check the relevant one :)
Let me know

**Comment by Smileyvero** [comment]
[QZ.zip](https://github.com/user-attachments/files/17646737/QZ.zip)

Yes, I'm absolutely sure it was Monday 28.10, it's registered both on Zwift and on Strava :) It was around 12-12.30. Run just over 5 km. Think I got all the files sent in one file now... That's all I have from that day. Hope you find the right file! :)

**Comment by cagnulein** [comment]
yeah so the previous one was from that timeframe but there are no metrics
at all from the treadmill. how do you pair it on zwift so? :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 6 nov 2024 alle ore 13:10 Smileyvero ***@***.***>
ha scritto:

**Comment by Smileyvero** [comment]
[QZ.zip](https://github.com/user-attachments/files/17647919/QZ.zip)

Maybe it was when I switched to using the Garmin watch. It works very badly, but it has been an emergency solution... But now I have arranged a brand new debug log. I connected to the mill with Zwift, used the QZ app as a bridge. It works until about 6 minutes into the race, then bluetooth is switched off. Then I close Zwift and the QZ app, when I open QZ again, it connects to the treadmill again. But then it disconnects once again (probably because the mill doesn't run, it has to run for the app to connect)

(I have to adjust both speed and incline manually.)

**Comment by cagnulein** [comment]
i have a good news, the treadmill is perfectly connected to qz. so when you are saying "bluetooth is switched off" do you mean on zwift? on which device zwift is running? on which qz? did you try to run zwift and qz on the same device enabling the android notification icon setting in the experimental settings in qz?

**Comment by Smileyvero** [comment]
So good to hear!! :) 

It's the bluetooth on the treadmill that switches off, and then everything on Zwift also stops. The runner (me) on Zwift stops, and everything on Zwift stops working 🤔
I use Zwift on my mobile (android), haven't tested what you suggest, so I can test that right now😄

**Comment by cagnulein** [comment]
it's strange because since to the last line of the log the treadmill was correctely connected to qz. anyway give it a go and let me know

**Comment by Smileyvero** [comment]
[QZ.zip](https://github.com/user-attachments/files/17648523/QZ.zip)

Incline settings on QZ:

![Screenshot_20241106_153030](https://github.com/user-attachments/assets/e9d02d70-61e1-4c7d-af56-2b71c2c27f0e)
![Screenshot_20241106_153004](https://github.com/user-attachments/assets/c155cf32-5482-4722-88a2-8009058562be)

**Comment by cagnulein** [comment]
ok it's not qz writing 0 to the incline. qz is not sending any command at all. do you see this problem even without qz? or with the sole app?

**Comment by Smileyvero** [comment]
Tested Zwift with QZ again. This time I tried to control the incline with QZ, and not with the treadmill. It didn't work either, the incline went down to 0 again after a short time. Then I tried to control the treadmill (incline and speed) with the Sole app, it worked. Have had hill sessions on the treadmill before, without problem :) But it has been without a bluetooth connection.

**Comment by cagnulein** [comment]
Ok I will check the log again finding any issue

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 6 nov 2024 alle 18:03 Smileyvero ***@***.***>
ha scritto:

**Comment by Smileyvero** [comment]
[QZ.zip](https://github.com/user-attachments/files/17656943/QZ.zip)

Now I tested incline on the treadmill, only with the QZ app connected to bluetooth. Used the app to control the mill. But the same thing happened: after a little while, the mill goes down again. Attaching a new debug log :)

**Comment by Smileyvero** [comment]
Great, just take a look at it when you have time, no stress :) I'm just doing some testing on my own for so long.

**Comment by cagnulein** [comment]
@Smileyvero Didn't forget about you, I'm just overwhelmed by tickets

**Comment by Smileyvero** [comment]
Thanks for the message, I understand you have a lot to do, no problem. :)
